src/iceberg/manifest_writer.h

```diff
@@ -54,9 +69,15 @@ class ICEBERG_EXPORT ManifestWriter {
   /// \brief Close writer and flush to storage.
   Status Close();
 
+  /// \brief Get the metrics of written manifest file.
+  std::optional<Metrics> metrics() const;
+
   /// \brief Get the content of the manifest.
   ManifestContent content() const;
 
+  /// \brief Get the ManifestFile object.
+  Result<ManifestFile> ToManifestFile() const;
```

Comment:
I think we should be consistent for `metrics()` and `ToManifestFile()`. What about using the following:

```cpp
Result<Metrics> metrics() const;
Result<ManifestFile> ToManifestFile() const;  // unchanged
```

We just return invalid state error when file is not closed.

Reply:
Perhaps we should also change `FileWriter::metrics` to return `Result<Metrics>` as well and return not supported error for some writers.